### PR TITLE
Do not try to reinterview deleted node

### PR DESCRIPTION
**Pull request by marcelveldt**
Handle a minor edge case where we keep retrying to interview an already deleted node.

### CODE REVIEW

File: matter_server/server/device_controller.py [hunk 1]
```diff
@@ -762,6 +762,10 @@ async def _check_interview_and_subscription(
     ) -> None:
         """Handle interview (if needed) and subscription for known node."""
 
+        if node_id not in self._nodes:
+            # edge case: this may happen if the node was deleted while in retrying state
```
[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/server/device_controller.py:766]: Yes, that is what we do and this handles the race condition from the split moment in between

[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/server/device_controller.py:766]: It is really an edge case but still I ran into it once.

[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/server/device_controller.py:766]: We could also store the TimerHandle into an instance variable but maybe that is overcomplicating it for such an edge case while this does the job just fine ?

[the remark this answers is not in the document]
Comment (marcelveldt) [reply matter_server/server/device_controller.py]: Decided to adjust it anyway.

**Review by frenck — APPROVED** [review]

**Pull request merged**